plugin/src/main/java/org/opensearch/ml/action/connector/DeleteConnectorTransportAction.java

```diff
@@ -58,92 +68,158 @@ public DeleteConnectorTransportAction(
-            if (Boolean.TRUE.equals(x)) {
-                try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-                    SearchRequest searchRequest = new SearchRequest(ML_MODEL_INDEX);
-                    SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
-                    sourceBuilder.query(QueryBuilders.matchQuery(MLModel.CONNECTOR_ID_FIELD, connectorId));
-                    searchRequest.source(sourceBuilder);
-                    client.search(searchRequest, ActionListener.runBefore(ActionListener.wrap(searchResponse -> {
-                        SearchHit[] searchHits = searchResponse.getHits().getHits();
-                        if (searchHits.length == 0) {
-                            deleteConnector(deleteRequest, connectorId, actionListener);
-                        } else {
-                            log
-                                .error(
-                                    searchHits.length + " models are still using this connector, please delete or update the models first!"
-                                );
-                            List<String> modelIds = new ArrayList<>();
-                            for (SearchHit hit : searchHits) {
-                                modelIds.add(hit.getId());
-                            }
-                            actionListener
-                                .onFailure(
-                                    new OpenSearchStatusException(
-                                        searchHits.length
-                                            + " models are still using this connector, please delete or update the models first: "
-                                            + Arrays.toString(modelIds.toArray(new String[0])),
-                                        RestStatus.CONFLICT
-                                    )
-                                );
-                        }
-                    }, e -> {
-                        if (e instanceof IndexNotFoundException) {
-                            deleteConnector(deleteRequest, connectorId, actionListener);
-                            return;
-                        }
-                        log.error("Failed to delete ML connector: " + connectorId, e);
-                        actionListener.onFailure(e);
-                    }), context::restore));
-                } catch (Exception e) {
-                    log.error(e.getMessage(), e);
-                    actionListener.onFailure(e);
-                }
-            } else {
-                actionListener.onFailure(new MLValidationException("You are not allowed to delete this connector"));
+        String tenantId = mlConnectorDeleteRequest.getTenantId();
+        tenantAwareHelper.setTenantId(tenantId);
+        if (!tenantAwareHelper.validateTenantId(actionListener))
+            return;
+
+        connectorAccessControlHelper
+            .validateConnectorAccess(
```

Comment:
Why not merge `checkConnectorPermission` to this `validateConnectorAccess` method?

Reply:
Yeah, I have plan to do that, but if I touch that file, then I need to make more changes to register model, update model (multiple files). Just to keep the scope of the PR small, I would like to address that in the follow up PR when I'll make changes to the model apis.

Reply:
Sounds good, you can add some TODO in code